pkg/aws/account_provider.go

```diff
@@ -101,5 +101,5 @@ func (a *accountProvider) NewStorage(cfg *config.Storage) (resource.ProviderStor
 }
 
 func (a *accountProvider) NewBucket(b resource.Bucket, cfg *config.Bucket) (resource.ProviderBucket, error) {
-	return newBucket(b, cfg, a.s3[cfg.Region()])
+	return newBucket(b, cfg, a.s3[cfg.Region()], a)
```

Comment:
Since you are passing in the accountProvider and this cfg, you don't need to pass in s3.